warehouse/query-core/src/main/java/datawave/query/util/AttributeComparator.java

```diff
@@ -0,0 +1,195 @@
+import datawave.query.attributes.Attributes;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+public final class AttributeComparator {
+    
+    // this is a utility and should never be instantiated
+    private AttributeComparator() {
+        throw new UnsupportedOperationException();
+    }
+    
+    /**
+     * Performs a simple comparison on a set of Attribute's data and some parts of their metadata
+     *
+     * @param attr1
+     *            An Attribute against which we want to compare
+     * @param attr2
+     *            Another Attribute against which we want to check for likeness
+     * @return
+     */
+    public static boolean singleToSingle(final Attribute attr1, final Attribute attr2) {
+        return (attr1.getData().equals(attr2.getData()) && attr1.getColumnVisibility().equals(attr2.getColumnVisibility()) && attr1.getTimestamp() == attr2
+                        .getTimestamp());
+    }
+    
+    /**
+     * Performs a simple comparison on a set of Attribute's data and some parts of their metadata
+     *
+     * @param attr
+     *            An Attribute against which we want to compare
+     * @param attrs
+     *            An Attributes against which we want to check for any matches contained therein
+     * @return
+     */
+    public static boolean singleToMultiple(final Attribute attr, final Attributes attrs) {
+        return attrs.getAttributes().stream().anyMatch(existingAttribute -> singleToSingle(existingAttribute, attr));
+    }
+    
+    /**
+     * Performs a simple comparison on a set of Attribute's data and some parts of their metadata
+     *
+     * @param attrs1
+     *            An Attributes against which we want to compare
+     * @param attrs1
+     *            An Attributes against which we want to check for any matches contained between the two sets
+     * @return
+     */
+    public static boolean multipleToMultiple(final Attributes attrs1, final Attributes attrs2) {
+        boolean containsMatch = false;
+        for (Attribute<? extends Comparable<?>> newAttr : attrs2.getAttributes()) {
+            if (singleToMultiple(newAttr, attrs1)) {
+                containsMatch = true;
+            }
+        }
+        return containsMatch;
+    }
+    
+    /**
+     * Performs a basic match on the content/data of two given Attributes. If such a match is found, combine the metadata from the two Attributes and create new
+     * deduplicated Attributes to return to the Document. Preference is given to the Attribute found in the event, NOT the index. In the case that true
+     * duplicates are discovered we keep the Attribute that already exists within the Document, set the Attribute's toKeep() flag to true (if not already), and
+     * ignore the potential new Attribute being checked.
```

Comment:
If everything else is equal, is priority given to first collection?

Reply:
Yes, that's correct - shall I add a note about that in the docs?

Reply:
Comment added to docs.